src/llama-hparams.h

```diff
@@ -110,17 +110,42 @@ struct llama_hparams {
     std::array<bool, LLAMA_MAX_LAYERS> swa_layers;
 
     // for State Space Models
-    uint32_t ssm_d_conv  = 0;
-    uint32_t ssm_d_inner = 0;
-    uint32_t ssm_d_state = 0;
-    uint32_t ssm_dt_rank = 0;
-    uint32_t ssm_n_group = 0;
+    uint32_t ssm_d_conv     = 0;
+    uint32_t ssm_d_inner    = 0;
+    uint32_t ssm_d_state    = 0;
+    uint32_t ssm_dt_rank    = 0;
+    uint32_t ssm_n_group    = 0;
+    bool     ssm_dt_b_c_rms = false;
+    uint32_t ssm_head_dim   = 0;
+    uint32_t ssm_mamba_d_ssm = 0;
+
+    // Falcon-H1 specific parameters
```

Comment:
This is definitely interesting. My guess is that there will be some pushback on adding this many new `hparams` to the central struct that are specific to a single architecture. Many of these appear to have analogous params elsewhere in the struct, so the general preference is to reuse existing params (even if their names don't match with the HF names) and then do the name translation in the `conver_hf_to_gguf.py` / `gguf-py` portion.